modules/src/ics02_client/client_def.rs

```diff
@@ -119,11 +122,14 @@ pub enum AnyClientState {
     Mock(MockClientState),
 }
 
-impl AnyClientState {
-    pub fn from_any(any: prost_types::Any) -> Result<Self, Error> {
-        match any.type_url.as_str() {
+impl TryFromRaw for AnyClientState {
+    type RawType = prost_types::Any;
+    type Error = Error;
+
+    fn try_from(value: Self::RawType) -> Result<Self, Self::Error> {
+        match value.type_url.as_str() {
             "ibc.tendermint.ClientState" => {
-                let raw = RawTendermintClientState::decode(any.value.as_ref())
+                let raw = RawTendermintClientState::decode(value.value.as_ref())
```

Comment:
It was necessary for `AnyClientState` to implement the `TryFromRaw` trait, otherwise the call below would fail:

https://github.com/informalsystems/ibc-rs/blob/106c459cab6cc1a6359759c7bfc2d0d62e004686/relayer-cli/src/commands/query/client.rs#L80-L81

The failure occurs because the function `chain.query::<T>` requires the trait bound: `T: TryFromRaw`.

> Also value.value.as_ref() doesn't ready very well :)

Agree! Didn't notice that, will fix.